### PR TITLE
added compressed remote GPS message for limited datalinks

### PR DESCRIPTION
**Pull request by kirkscheper**
Re-purposed the WC_RSSI message (which appeared not be be used).
updated remote_gps to be fowarded.
added a very small gps message and a swarming message.

### CODE REVIEW

File: message_definitions/v1.0/messages.xml [hunk 1]
```diff
@@ -1997,7 +1999,24 @@
     <field name="imageBuffer" type="uint8[]"/>
   </message>
 
-  <!--230 is free -->
+  <message name="SWARMNN" id="230">
```
Comment (flixr) [inline message_definitions/v1.0/messages.xml]: Can you please add descriptions (for the message as well as the fields) and add units?

File: message_definitions/v1.0/messages.xml [hunk 2]
```diff
@@ -2223,6 +2248,7 @@
   <field name="utm_east"  type="int32" unit="cm"></field>
   <field name="utm_north"  type="int32" unit="cm"></field>
   <field name="alt"    type="int32" unit="cm">Height above Mean Sea Level (geoid)</field>
+  <field name="utm_zone"  type="int16"></field>
```
Comment (flixr) [inline message_definitions/v1.0/messages.xml]: why make the zone an int16_t instead of uint8_t?

File: message_definitions/v1.0/messages.xml [hunk 3]
```diff
@@ -2223,6 +2248,7 @@
   <field name="utm_east"  type="int32" unit="cm"></field>
   <field name="utm_north"  type="int32" unit="cm"></field>
   <field name="alt"    type="int32" unit="cm">Height above Mean Sea Level (geoid)</field>
+  <field name="utm_zone"  type="int8"></field>
```
Comment (flixr) [inline message_definitions/v1.0/messages.xml]: still think this should be `uint8_t`

**Comment by flixr** [comment]
validation failed, please fix wrong field close tags

**Comment by kirkscheper** [comment]
I get a bunch of unaligned message errors when I make the messages in a test (in messages that are already in the code). Shall I move the message items around to correct that or is it not an issue?

**Comment by flixr** [comment]
what test and what messages?

**Comment by kirkscheper** [comment]
never mind, ends up that the intermcu is not possible to be aligned

**Comment by flixr** [comment]
There is also too much stuff in one commit/PR that makes it really hard to merge only some of your other pull requests...
see remote_gps_small branch, do you really want the ac_id removed?

**Comment by kirkscheper** [comment]
I don't want the ac_id removed, I just moved it to the end of the message to stop the not aligned error when that is important.

The branch looks good, pull it!.

**Comment by flixr** [comment]
For the RSSI message, should it get some field to indicate where the RSSI is from?

**Comment by kirkscheper** [comment]
Yea, I guess you could do that, I use it now to communicate the rssi
received at the Bluetooth dongle to the drone so the acid is in the message
sender ac_id. But yea, of course if you use the message as a downlink then
the explicit ID is interesting, haha. I will add it.

Kirk

**Comment by flixr** [comment]
I was more thinking about different components on the aircraft reporting RSSI, but if we don't need that right now, maybe we better wait until we add a component id to all messages?

File: message_definitions/v1.0/messages.xml [hunk 4]
```diff
@@ -2187,8 +2205,14 @@
...
 
-  <!-- 251 is free -->
-  <!-- 252 is free -->
+   <!--251 is free -->
+
+   <message name="GPS_SMALL" id="252">
+    <field name="multiplex_speed"  type="uint32">bits 31-21 course in decideg : bits 20-10 ground speed in cm/s : bits 9-0 climb speed in cm/s</field>
+    <field name="lla_lat"  type="int32" unit="1e7deg" alt_unit="deg" alt_unit_coef="0.0000001"></field>
+    <field name="lla_lon"  type="int32" unit="1e7deg" alt_unit="deg" alt_unit_coef="0.0000001"></field>
+    <field name="alt"    type="int16" unit="cm" alt_unit="m" alt_unit_coef="0.01"></field>
```
Comment (flixr) [inline message_definitions/v1.0/messages.xml]: This is supposed to be alt above MSL(geoid) or ellipsoid?

**Comment by flixr** [comment]
Added the RSSI message with ab69fa785b6145037c949873d92a9aed65160949

**Comment by kirkscheper** [comment]
I removed the swarm message for now, in the end I am not sure this will be useful for others. The compressed message may be useful though...

File: message_definitions/v1.0/messages.xml [hunk 5]
```diff
@@ -2208,8 +2208,14 @@
       <field name="yaw_c"  type="float"/>
   </message>
 
-  <!-- 251 is free -->
-  <!-- 252 is free -->
+   <!--251 is free -->
+
+   <message name="GPS_SMALL" id="252">
+    <field name="multiplex_speed"  type="uint32">bits 31-21 course in decideg : bits 20-10 ground speed in cm/s : bits 9-0 climb speed in cm/s</field>
+    <field name="lla_lat"  type="int32" unit="1e7deg" alt_unit="deg" alt_unit_coef="0.0000001"></field>
```
Comment (flixr) [inline message_definitions/v1.0/messages.xml]: as already mentioned in the other PR, I think it makes sense to keep the `lat` and `lon` field names like in all the other messages or is there a reason for naming them `lla_lat` and `lla_lon` here?

Same for ACINFO_LLA

File: message_definitions/v1.0/messages.xml [hunk 6]
```diff
@@ -2208,8 +2208,14 @@
...
 
-  <!-- 251 is free -->
-  <!-- 252 is free -->
+   <!--251 is free -->
+
+   <message name="GPS_SMALL" id="252">
+    <field name="multiplex_speed"  type="uint32">bits 31-21 course in decideg : bits 20-10 ground speed in cm/s : bits 9-0 climb speed in cm/s</field>
+    <field name="lat"  type="int32" unit="1e7deg" alt_unit="deg" alt_unit_coef="0.0000001"></field>
+    <field name="lon"  type="int32" unit="1e7deg" alt_unit="deg" alt_unit_coef="0.0000001"></field>
+    <field name="alt"    type="int16" unit="cm" alt_unit="m" alt_unit_coef="0.01">height above the ellipsoid</field>
```
Comment (flixr) [inline message_definitions/v1.0/messages.xml]: btw, `alt_unit_coef` is not needed for the ones listed in message_definitions/common/units.xml

File: message_definitions/v1.0/messages.xml [hunk 7]
```diff
@@ -2353,6 +2360,17 @@
    <field name="length" type="uint8"/>
    <field name="rtcm" type="uint8[]"/>
  </message>
+ 
+  <message name="ACINFO_LLA" id="14" link="broadcasted">
+  <field name="course" type="int16" unit="decideg"></field>
+  <field name="lat"  type="int32" unit="1e7deg" alt_unit="deg" alt_unit_coef="0.0000001"></field>
+  <field name="lon"  type="int32" unit="1e7deg" alt_unit="deg" alt_unit_coef="0.0000001"></field>
+  <field name="alt"    type="int32" unit="cm">Height above ellipsoid</field>
```
Comment (flixr) [inline message_definitions/v1.0/messages.xml]: Probably nice to add SI units as `alt_unit` so in this case `alt_unit="m"`.
If we add that to all relevant fields, it makes comparing and plotting much easier.

File: message_definitions/v1.0/messages.xml [hunk 8]
```diff
@@ -2543,10 +2561,10 @@
...
 
   <message name="REMOTE_GPS_SMALL" id="54" link="forwarded">
-    <field name="heading"   type="int16"  unit="rad*1e4"/>
-    <field name="pos_xyz"  	type="uint32" unit="cm">bits 31-21 x position in cm : bits 20-10 y position in cm : bits 9-0 z position in cm</field>
-    <field name="speed_xyz" type="uint32" unit="cm/s">bits 31-21 speed x in cm/s : bits 20-10 speed y in cm/s : bits 9-0 speed z in cm/s</field>
-    <field name="tow"   	type="uint32" unit="ms" alt_unit="m"/>
+    <field name="heading"   type="int16"  unit="1e4rad" alt_unit="rad" alt_unit_coef="0.0001"/>
+    <field name="pos_xyz"  	type="uint32">bits 31-21 x position in cm : bits 20-10 y position in cm : bits 9-0 z position in cm</field>
+    <field name="speed_xyz" type="uint32">bits 31-21 speed x in cm/s : bits 20-10 speed y in cm/s : bits 9-0 speed z in cm/s</field>
+    <field name="tow"   	type="uint32" unit="ms" alt_unit="s" alt_unit_coef="1000."/>
```
Comment (flixr) [inline message_definitions/v1.0/messages.xml]: alt_unit_coef should be 0.001

File: message_definitions/v1.0/messages.xml [hunk 9]
```diff
@@ -2543,10 +2561,10 @@
   </message>
 
   <message name="REMOTE_GPS_SMALL" id="54" link="forwarded">
-    <field name="heading"   type="int16"  unit="rad*1e4"/>
-    <field name="pos_xyz"  	type="uint32" unit="cm">bits 31-21 x position in cm : bits 20-10 y position in cm : bits 9-0 z position in cm</field>
-    <field name="speed_xyz" type="uint32" unit="cm/s">bits 31-21 speed x in cm/s : bits 20-10 speed y in cm/s : bits 9-0 speed z in cm/s</field>
-    <field name="tow"   	type="uint32" unit="ms" alt_unit="m"/>
+    <field name="heading"   type="int16"  unit="1e4rad" alt_unit="rad" alt_unit_coef="0.0001"/>
```
Comment (flixr) [inline message_definitions/v1.0/messages.xml]: alt_unit would be nicer in deg

**Comment by flixr** [comment]
LGTM
@gautierhattenberger fine to merge it?

**Pull request merged**